plugins/modules/azure_rm_adgroup.py

```diff
@@ -0,0 +1,409 @@
+        state: 'present'
+        absent_members:
+          - "{{ ad_object_1_object_id }}"
+
+    - name: Ensure Users are Members of a Group using object_id
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: 'present'
+        absent_members:
+          - "{{ ad_object_1_object_id }}"
+
+    - name: Ensure Users are Owners of a Group using display_name and mail_nickname
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        display_name: "Group-Name"
+        mail_nickname: "Group-Mail-Nickname"
+        state: 'present'
+        present_owners:
+          - "https://graph.windows.net/{{ tenant_id }}/directoryObjects/{{ ad_object_1_object_id }}"
+          - "https://graph.windows.net/{{ tenant_id }}/directoryObjects/{{ ad_object_2_object_id }}"
+
+    - name: Ensure Users are Owners of a Group using object_id
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: 'present'
+        present_owners:
+          - "https://graph.windows.net/{{ ad_object_1_tenant_id }}/directoryObjects/{{ ad_object_1_object_id }}"
+          - "https://graph.windows.net/{{ ad_object_2_tenant_id }}/directoryObjects/{{ ad_object_2_object_id }}"
+
+    - name: Ensure Users are not Owners of a Group using display_name and mail_nickname
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        display_name: "Group-Name"
+        mail_nickname: "Group-Mail-Nickname"
+        state: 'present'
+        absent_owners:
+          - "{{ ad_object_1_object_id }}"
+          - "{{ ad_object_2_object_id }}"
+
+    - name: Ensure Users are Owners of a Group using object_id
+      azure.azcollection.azure_rm_adgroup:
+        tenant: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: 'present'
+        absent_owners:
+          - "{{ ad_object_1_object_id }}"
+          - "{{ ad_object_2_object_id }}"
+
+'''
+
+RETURN = '''
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_ext import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.graphrbac.models import GraphErrorException
+    from azure.graphrbac.models import GroupCreateParameters
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMADGroup(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            object_id=dict(type='str'),
+            display_name=dict(type='str'),
+            mail_nickname=dict(type='str'),
+            present_members=dict(type='list', elements='str'),
+            present_owners=dict(type='list', elements='str'),
```

Comment:
same as memebrs, should we combine two owner fields into one?

Reply:
@haiyuazhang see the comment above as it addresses both members and owners.